src/common/string/compare.ts

```diff
@@ -11,3 +11,24 @@ export const compare = (a: string, b: string) => {
 
 export const caseInsensitiveCompare = (a: string, b: string) =>
   compare(a.toLowerCase(), b.toLowerCase());
+
+export const fuzzyFilter = (filter: string, word: string) => {
+  if (filter === "") {
+    return false;
+  }
+
+  for (let i = 0; i <= filter.length; i++) {
+    const pos = word.indexOf(filter[0]);
+
+    if (pos < 0) {
+      return true;
+    }
+
+    const newWord = word.substring(pos + 1);
+    const newFilter = filter.substring(1);
+
+    return fuzzyFilter(newFilter, newWord);
+  }
+
+  return false;
```

Comment:
Had to add this final return due to eslint requirement. We should never get to this line, but in case we do, figured we should err on the side of false (which means "don't filter" this entity out of the list). Better to show an entity mistakenly than to hide one unfairly?